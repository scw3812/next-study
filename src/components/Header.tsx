import Link from 'next/link';
import React from 'react';
import tw from 'tailwind-styled-components';

const Header = () => {
  return (
    <div>
      <Link href="/">
        <HeaderAnchor>Home</HeaderAnchor>
      </Link>
      <Link href="/about">
        <HeaderAnchor>About</HeaderAnchor>
      </Link>
    </div>
  );
};

const HeaderAnchor = tw.button`
  mr-4
`;

export default Header;
