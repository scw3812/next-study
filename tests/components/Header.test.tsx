import { render, screen } from '@testing-library/react';
import Header from '../../src/components/Header';

describe('Header', () => {
  it('renders buttons', () => {
    const { container } = render(<Header />);

    const homeButton = screen.getByText('Home');
    const aboutButton = screen.getByText('About');

    expect(homeButton).toBeInTheDocument();
    expect(aboutButton).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
