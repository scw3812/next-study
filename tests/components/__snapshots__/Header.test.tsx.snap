// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders buttons 1`] = `
<div>
  <div>
    <button
      class="mr-4"
    >
      Home
    </button>
    <button
      class="mr-4"
    >
      About
    </button>
  </div>
</div>
`;
